src/edu/washington/escience/myria/operator/network/distribute/BroadcastDistributeFunction.java

```diff
@@ -0,0 +1,36 @@
+package edu.washington.escience.myria.operator.network.distribute;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import edu.washington.escience.myria.util.MyriaArrayUtils;
+
+/**
+ * Broadcast distribute function.
+ */
+public final class BroadcastDistributeFunction extends DistributeFunction {
+
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   */
+  @JsonCreator
+  public BroadcastDistributeFunction() {
+    super(new SinglePartitionFunction(), null);
+  }
+
+  /**
+   * @param numDestinations number of destination
+   */
+  public BroadcastDistributeFunction(final int numDestinations) {
```

Comment:
This brings me to another point. Using the current 1:1 identification of partitions with workers it's trivial to infer the mapping of partitions to workers. But if partitions are now mapped to "destinations" which are actually ephemeral system-issued IDs whose meaning can't be reconstructed after the fact, that seems impossible. We can still inspect the list of worker IDs in `HowDistributed` to get the workers storing the relation (e.g., to schedule fragments for execution on those workers), but I assume we would also want to be able to reconstruct the mapping of partitions to workers by inspecting the `DistributeFunction` serialized in the catalog. But how do we do that when this mapping is intermediated by an ephemeral layer (partitions->channel IDs) that can't be reconstructed after the fact? And why persist this ephemeral mapping at all (as JSON serialized in the catalog) if it can't be meaningfully interpreted?

An alternative, I think, would be use the partition->channel ID mapping during execution but to generate an explicit mapping of partitions->workers for JSON serialization in the catalog.